numba/cuda/cudadrv/devicearray.py

```diff
@@ -126,7 +126,8 @@ def __cuda_array_interface__(self):
             'strides': None if is_contiguous(self) else tuple(self.strides),
             'data': (ptr, False),
             'typestr': self.dtype.str,
-            'version': 2,
+            'stream': int(self.stream) if self.stream is not 0,
```

Comment:
I think this should be:
```suggestion
            'stream': int(self.stream) if self.stream != 0 else None,
```

But this raises the question of if Numba has the stream set to `0` does it synchronize after kernels where it would guarantee that the stream doesn't need to be synchronized?

Reply:
In Numba internals `0` means "no default stream" (with the Numba meaning of "default stream", a stream to assume for async operations where the stream is not specified). If it's the default stream 0, then it will be a `Stream` object with a handle of `0`, which will compare equal to `0`, which is why I was specifically checking `is not 0` (and relying on the interning of small integers in CPython). I need to think about this again to be confident in what I mean here.

Reply:
Turns out that `if self.stream != 0 else None` (@kkraus14's suggestion) does the right thing in Numba so I've applied that.